src/Cake.Common/Tools/VSWhere/VSWhereAliases.cs

```diff
@@ -0,0 +1,246 @@
+        {
+            var settings = new VSWhereAllSettings();
+            return VSAll(context, settings);
+        }
+
+        /// <summary>
+        /// Gets all Visual Studio product installation paths.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="settings">The settings.</param>
+        /// <returns>The Visual Studio installation paths.</returns>
+        /// <example>
+        /// <code>
+        ///     var latestInstallationPaths = VSAll(new VSWhereAllSettings { Requires = "'Microsoft.Component.MSBuild" });
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("All")]
+        [CakeNamespaceImport("Cake.Common.Tools.VSWhere.All")]
+        public static IEnumerable<string> VSAll(this ICakeContext context, VSWhereAllSettings settings)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            var all = new VSWhereAll(context.FileSystem, context.Environment, context.ProcessRunner, context.Tools);
+            return all.All(settings);
+        }
+
+        /// <summary>
+        /// Gets Visual Studio product installation paths.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <returns>The Visual Studio installation paths.</returns>
+        /// <example>
+        /// <code>
+        ///     var latestInstallationPaths = VSProducts();
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("Product")]
+        [CakeNamespaceImport("Cake.Common.Tools.VSWhere.Product")]
+        public static IEnumerable<string> VSProducts(this ICakeContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+            var settings = new VSWhereProductSettings();
+            var product = new VSWhereProduct(context.FileSystem, context.Environment, context.ProcessRunner, context.Tools);
+            return product.Products(settings);
+        }
+
+        /// <summary>
+        /// Gets Visual Studio product installation paths.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="products">The products to find.</param>
+        /// <returns>The Visual Studio installation paths.</returns>
+        /// <example>
+        /// <code>
+        ///     var latestInstallationPaths = VSProducts("BuildTools");
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("Product")]
+        [CakeNamespaceImport("Cake.Common.Tools.VSWhere.Product")]
+        public static IEnumerable<string> VSProducts(this ICakeContext context, string products)
```

Comment:
Wondering if `string products` shouldn't be `IEnumerable<string>` looking at their examples it seems to be a space separated string
```
set ids=%pre%Community %pre%Professional %pre%Enterprise %pre%BuildTools
vswhere -latest -products %ids%
```